### PR TITLE
workaroud for the flaky oidc e2e test

### PR DESCRIPTION
**Pull request by zhaohuabing**
Delete and recreate SecurityPolicy to force repushing xDS to Envoy to recover from the OIDC test failure. This is just a workaround to reduce the time of e2e test due to  the failed OIDC test. We still need to investage the root cause and fix it.

Related issue: https://github.com/envoyproxy/gateway/issues/3898
Release Notes: No

### CODE REVIEW

File: test/e2e/tests/oidc.go [hunk 1]
```diff
@@ -135,13 +136,25 @@ func testOIDC(t *testing.T, suite *suite.ConformanceTestSuite) {
...
+			if res.StatusCode != http.StatusOK {
+				tlog.Logf(t, "Failed to get the login page, expected 200 OK, got %d", res.StatusCode)
+				return false, nil
+			}
 
 			// Parse the response body to get the URL where the login page would post the user-entered credentials
-			if err := client.ParseLoginForm(res.Body, keyCloakLoginFormID); err != nil {
+			if err := oidcClient.ParseLoginForm(res.Body, keyCloakLoginFormID); err != nil {
 				tlog.Logf(t, "failed to parse login form: %v", err)
+				// restart the envoy proxy to recover from the error, this is a workaround for the flaky test
```
Comment (zirain) [inline test/e2e/tests/oidc.go:152]: add the issue link here

File: test/e2e/tests/oidc.go [hunk 2]
```diff
@@ -135,13 +136,25 @@ func testOIDC(t *testing.T, suite *suite.ConformanceTestSuite) {
...
+			}
 
 			// Parse the response body to get the URL where the login page would post the user-entered credentials
-			if err := client.ParseLoginForm(res.Body, keyCloakLoginFormID); err != nil {
+			if err := oidcClient.ParseLoginForm(res.Body, keyCloakLoginFormID); err != nil {
 				tlog.Logf(t, "failed to parse login form: %v", err)
+				// restart the envoy proxy to recover from the error, this is a workaround for the flaky test: https://github.com/envoyproxy/gateway/issues/3898
+				// TODO: we should investigate the root cause of the flakiness and remove this workaround
+				proxyLabel := map[string]string{"gateway.envoyproxy.io/owning-gateway-name": "same-namespace"}
+				err := suite.Client.DeleteAllOf(context.TODO(), &corev1.Pod{}, client.MatchingLabels(proxyLabel), client.InNamespace("envoy-gateway-system"))
```
Comment (zirain) [inline test/e2e/tests/oidc.go:155]: should we force a rollout/restart instead of deleting?

Comment (zhaohuabing) [reply test/e2e/tests/oidc.go:155]: It doesnt matter here. We just need to restart the Envoy pod to repush the xDS.

**Review by zirain — DISMISSED** [review]

File: test/e2e/tests/oidc.go (hunk 2 above)
Comment (arkodg) [reply test/e2e/tests/oidc.go:155]: Can we use another method to repush xDS, like delete and add config

**Review by zirain — APPROVED** [review]

**Review by arkodg — APPROVED** [review]

**Pull request merged**